### PR TITLE
[Lock] Replace LockInterface by LockFactory in DI

### LINKED ISSUE #14416: Deeprecate lock service
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#38576
| PR author(s) | @jderusse
| Merged in    | 5.x

### LINKED ISSUE #14425: [Lock][Semaphore] Add Factory::createFromKey and deprecate lock.store s…
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#38593
| PR author(s) | @jderusse
| Merged in    | 5.x

### PR DESCRIPTION
**Pull request by jderusse**
Fixes #14416 and fixes #14425

### CODE REVIEW

File: lock.rst [hunk 1]
```diff
@@ -193,6 +194,11 @@ To lock the default resource, autowire the lock using
     same lock, inject the ``LockFactory`` instead to create a separate lock
     instance for each service.
 
+    .. deprecated:: 5.2
+
+        The ``LockInterface`` service and ``lock`` alias are deprecated since
+        Symfony 5.2.
+
```
Comment (wouterj) [inline lock.rst]: I don't think this is needed, as we don't document it anymore. (and Symfony's deprecation system already does a good job educating people about deprecations & the required changes)

File: lock.rst [hunk 2]
```diff
@@ -273,18 +279,20 @@ provides :ref:`named lock <reference-lock-resources-name>`::
...
+- e.g. ``invoice`` can be injected automatically by naming the argument
+``$invoiceLockFactory`` and type-hinting it with
+:class:`Symfony\\Component\\Lock\\LockFactory`.
+
+.. deprecated:: 5.2
+
+    The services for ``LockInterface`` (ie. ``lock``, ``lock.invoice``,
+    ``LockInterface $invoiceLock``) and ``PersistingStoreInterface`` (ie.
+    ``lock.store``, ``lock.invoice.store``, ``PersistingStoreInterface
+    $invoiceLockStore``) are deprecated Symfony 5.2.
```
Comment (wouterj) [inline lock.rst]: same here

**Review by wouterj — APPROVED** [review]

File: lock.rst (hunk 1 above)
Comment (jderusse) [reply lock.rst]: removed

**Comment by javiereguiluz** [comment]
Thank you Jérémy.

**Pull request merged**